ckanext/oidc_pkce/plugin.py

```diff
@@ -40,10 +46,39 @@ def get_helpers(
     ):
         return helpers.get_helpers()
 
-    if not tk.check_ckan_version("2.10"):
-        p.implements(p.IAuthenticator, inherit=True)
+    # IAuthenticator
+
+    if tk.check_ckan_version("2.10"):
+        def logout(self):
+            """ We want to return a view after the regular logout logic,
+            rather than before.
+
+            We set a flag to indicate that we're in the middle of logout,
+            then call the regular logout view. The view calls a second
+            instance of this function, which detects the flag and no-ops,
+            allowing the view to proceed and wipe the session.
 
-        # IAuthenticator
+            After it completes, we assemble a redirect and pass that back
+            to the code that originally called this function.
+            """
+            if session.pop("_in_logout", False):
```

Comment:
should we not be checking to see that the user is 'sso' logged in prior to doing all this work (i.e. if not sso logged in you can return none and let the standard logout do its thing.

Reply:
I'm not sure we have a way to determine whether or not an account is linked to SSO.

Reply:
There is a helper to tell you if they are an sso user. (question is if that is per login or set once sso used)

see https://github.com/qld-gov-au/ckanext-oidc-pkce/blob/master/ckanext/oidc_pkce/helpers.py#L20C5-L20C26 
```
def oidc_pkce_is_sso_user(id_or_name: str) -> bool:
...
extras = user["plugin_extras"] or {}
    return "oidc_pkce" in extras
```

Reply:
That does look good, _but_ the plugin already has a suitable user object at the relevant point, so adding an extra API call seems unnecessary.